client/shipping-label/state/actions.js

```diff
@@ -70,18 +99,29 @@ export const purchaseLabel = () => ( dispatch, getState, { callbackURL, nonce, s
 		}
 	};
 
-	const form = getState().shippingLabel.form;
-	const formData = {
-		origin: form.origin.values,
-		destination: form.destination.values,
-		payment_method_id: '123456789',
-		carrier: 'usps',
-		packages: form.packages.values.map( ( pckg, index ) => ( {
-			...omit( pckg, [ 'items', 'id' ] ),
-			service_id: form.rates.values[ index ],
-			products: flatten( pckg.items.map( ( item ) => fill( new Array( item.quantity ), item.product_id ) ) ),
-		} ) ),
-	};
+	let form = getState().shippingLabel.form;
+	const addressesValidation = [];
+	if ( ! form.origin.isValidated ) {
+		addressesValidation.push( validateAddress( dispatch, form.origin.values, 'origin', addressValidationURL, nonce ) );
+	}
+	if ( ! form.destination.isValidated ) {
+		addressesValidation.push( validateAddress( dispatch, form.destination.values, 'destination', addressValidationURL, nonce ) );
+	}
+
+	Promise.all( addressesValidation ).then( () => {
+		form = getState().shippingLabel.form;
+		const formData = {
+			origin: form.origin.pickNormalized ? form.origin.normalized : form.origin.values,
+			destination: form.destination.pickNormalized ? form.destination.normalized : form.destination.values,
+			payment_method_id: '123456789',
+			carrier: 'usps',
```

Comment:
Please add an issue to save the carrier (ID) for the customer selected real-time rate to order meta during checkout (or fetch it from order meta instead of hardcoding here if it is being saved already.)